src/client/dfs/dfs.c

```diff
@@ -2165,10 +2218,13 @@ dfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, dfs_t **_dfs)
 	if (get_daos_obj_mode(flags) == -1)
 		return EINVAL;
 
-	prop = daos_prop_alloc(0);
+	prop = daos_prop_alloc(num_props);
```

Comment:
You could probably do something like the following which would be nicer as it's automatically the correct size.
```suggestion
	prop = daos_prop_alloc(ARRAY_SIZE(props));
```

Reply:
Good idea. I think this will be a bit more foolproof in the long run.